### PR TITLE
Mal tester integreation

### PR DESCRIPTION
**Pull request by psockoN7**

### CODE REVIEW

File: src/asn1acn.cpp [hunk 1]
```diff
@@ -240,6 +252,35 @@ void Asn1AcnPlugin::addBuildICDToProjectMenu()
...
+            &ProjectExplorer::SessionManager::startupProjectChanged,
+            [action](ProjectExplorer::Project *project) {
+                action->setEnabled(project != nullptr);
+                action->setParameter(project == nullptr ? QString() : project->displayName());
+            });
+
+    connect(action, &QAction::triggered, [this]() {
+        if (m_testGeneratorDialog.isNull()) {
+            auto paramsProvider = std::make_shared<TestGenerator::TestGeneratorParamsProvider>(
+                Settings::load<Settings::TestGenerator>());
```
Comment (hcorg) [inline src/asn1acn.cpp]: maybe it should get settings loaded earlier (in function argument)?

File: src/asn1acn.h [hunk 2]
```diff
@@ -81,8 +87,12 @@ private slots:
     void addToToolsMenu(ActionContainer *container);
 
     QPointer<Libraries::Wizard::ImportComponentWizard> m_importComponentWizard;
+
```
Comment (hcorg) [inline src/asn1acn.h]: unnecessary \n (or one after the next line is needed :)) it makes "test generator" grouped with "find usages".

File: src/test-generator/testgeneratorparamsdialog.cpp [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+#include "testgeneratorparamsdialog.h"
+#include "testgeneratorparamswidget.h"
+
+#include "parseddatastorage.h"
+
+using namespace Asn1Acn::Internal;
+using namespace Asn1Acn::Internal::TestGenerator;
+
+TestGeneratorParamsDialog::TestGeneratorParamsDialog(TestGeneratorParamsProviderPtr provider,
```
Comment (hcorg) [inline src/test-generator/testgeneratorparamsdialog.cpp]: is this provider used anywhere else apart from runner? I just wonder whether it is not a little overkill :) (plain TGParams passed to ->run(params) may be good enough)

Comment (psockoN7) [reply src/test-generator/testgeneratorparamsdialog.cpp]: The idea here is to have shared object (TgParamsProvider) between TestGeneratorsDialog and TestGeneratorRunner. Former sets params, latter reads them.

File: src/asn1acn.cpp (hunk 1 above)
Comment (psockoN7) [reply src/asn1acn.cpp]: I will fix this

File: src/asn1acn.h (hunk 2 above)
Comment (psockoN7) [reply src/asn1acn.h]: Will correct it.

**Pull request merged**